integration/command.go

```diff
@@ -82,32 +82,39 @@ var waitUntilInspektorGadgetPodsDeployed *command = &command{
 	done`,
 }
 
-var deploySPO *command = &command{
-	name: "Deploy Security Profiles Operator (SPO)",
-	// The security-profiles-operator-webhook deployment is not part of the
-	// yaml but created by SPO. We cannot use kubectl-wait before it is
-	// created, see also:
-	// https://github.com/kubernetes/kubernetes/issues/83242
+func deploySPO(limit bool) *command {
+	limitResourcesUsage := ""
+	if limit {
+		limitResourcesUsage = "sed 's/replicas: 3/replicas: 1/'|grep -v cpu: |"
+	}
+
+	// The security-profiles-operator-webhook deployment is not part of the yaml
+	// but created by SPO. We cannot use kubectl-wait before it is created, see
+	// also: https://github.com/kubernetes/kubernetes/issues/83242
 	// Unfortunately, it takes quite a while for
-	// security-profiles-operator-webhook to be started, hence the long
-	// timeout
-	cmd: `
-	kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.7.2/cert-manager.yaml
-	kubectl --namespace cert-manager wait --for condition=ready pod -l app.kubernetes.io/instance=cert-manager
-	curl https://raw.githubusercontent.com/kubernetes-sigs/security-profiles-operator/v0.4.2/deploy/operator.yaml | \
-		sed 's/replicas: 3/replicas: 1/'|grep -v cpu: | \
-		kubectl apply -f -
-	for i in $(seq 1 120); do
-		if [ "$(kubectl get pod -n security-profiles-operator -l app=security-profiles-operator,name=security-profiles-operator-webhook -o go-template='{{len .items}}')" -ge 1 ] ; then
-			break
-		fi
-		sleep 1
-	done
-	kubectl patch ds -n security-profiles-operator spod --type=json \
-		-p='[{"op": "remove", "path": "/spec/template/spec/containers/0/resources"}, {"op": "remove", "path": "/spec/template/spec/containers/1/resources"}, {"op": "remove", "path": "/spec/template/spec/initContainers/0/resources"}]'
-	kubectl --namespace security-profiles-operator wait --for condition=ready pod -l app=security-profiles-operator || (kubectl get pod -n security-profiles-operator ; kubectl get events -n security-profiles-operator ; false)
-	`,
-	expectedRegexp: "pod/security-profiles-operator-.*-.* condition met",
+	// security-profiles-operator-webhook to be started, hence the long timeout
+	cmdStr := fmt.Sprintf(`
+kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.7.2/cert-manager.yaml
+kubectl --namespace cert-manager wait --for condition=ready pod -l app.kubernetes.io/instance=cert-manager
+curl https://raw.githubusercontent.com/kubernetes-sigs/security-profiles-operator/v0.4.2/deploy/operator.yaml | \
+  %s \
+  kubectl apply -f -
```

Comment:
I wonder if it would be more readable if the Go code and bash code were not interleaved like this. Something like:
```
curl https://... | \
if [ %v = true ] ; then 
  sed 's/replicas: 3/replicas: 1/'|grep -v cpu:
fi | \
kubectl apply -f -
```
When fmt.Sprintf() is given a boolean like `limit` with %v, it renders the string "true" or "false".

What do you think?

Reply:
Yeah, I think it makes the code more readable.